src/app/app.component.ts

```diff
@@ -48,11 +49,19 @@ export class AppComponent implements OnDestroy {
         }
       });
 
-    if (this.configService.config?.oidcConfig) {
-      this.oidcService.checkAuth().pipe(
-        filter((d) => d.isAuthenticated),
-        map((d: LoginResponse) => fromAuth.oidcLogin(d)),
-      ).subscribe((action) => this.store.dispatch(action));
+    const err = new URLSearchParams(window.location.search).get('error');
+    if (err === "interaction_required") {
+      this.store.dispatch(fromAuth.oidcAutoLoginFailure());
+    } else if (this.configService.config?.oidcConfig) {
+      this.oidcService.checkAuth().subscribe((lr: LoginResponse) => {
+        if (lr.isAuthenticated) {
+          this.store.dispatch(fromAuth.oidcLogin(lr));
+        } else {
+          this.store.select(authFeatureSelector).pipe(
+            filter((feature) => !feature.autoLoginFailed)
+          ).subscribe(() => this.oidcService.authorize(undefined, {customParams:{prompt: 'none'}}))
```

Comment:
This is a nested subscription, I am not an expert but I believe this should be avoided, maybe use something like a filter like `filter((lr: LoginResponse) => !lr.isAuthenticated)` .
I think you should be able to only have one `else if` statement, with filtering, and run the `.subscribe()` at the end after filtering.

Reply:
@lanseg, do we need to access `autoLoginFailed` in other parts of the application?
If not have you considered using a private private field to store its value?

Reply:
Done. It turned into just a single finite subscription after putting autoLoginFailure in the static variable.